kafka-ui-react-app/src/components/Topics/Topic/Messages/Filters/Filters.tsx

```diff
@@ -204,7 +204,7 @@ const Filters: React.FC<FiltersProps> = ({
       q:
         queryType === MessageFilterType.GROOVY_SCRIPT
           ? activeFilter.code
-          : query,
+          : searchParams.get('q') || '',
```

Comment:
`query`  job is actually to save the state of the `searchParams.get('q')` maybe we have issue of updating the `query`.

Reply:
It is deserved, because 'Clear all' button (with `setQuery('')`) has no effect on `q` too.

maybe line188 to be updated?
`const [query, setQuery] = React.useState<string>(searchParams.get('q') || '');`

Reply:
first, the related commit may be https://github.com/provectus/kafka-ui/commit/049b35fc99fd09d270fbccb3f0a2bbcc9bf11f56 adding to https://github.com/provectus/kafka-ui/pull/2462
and i find more that.. searching messages with pagination is not working for the same reason now.


i guess **Search** component is related to not updating the `query`, but i'm not sure of side effects by fixing it because it is a common component.

so for now, my idea is still that replacing `query` with `searchParams.get('q') || ''` is appropriate to get the updated parameter.

Or use `setSearchParams` instead of `useState`? but i'm not a expert, so needs to be reviewed

Reply:
hi, can you review it? @Haarolean 
the main problem is `q` is initiated when other parameters are changed.
so i fixed it by getting the updated value with `searchParms`, same as keySerde, valueSerde.
but to solve the root cause like @Mgrdich said, i think **Search** needs to be updated.